src/which.js

```diff
@@ -13,13 +13,25 @@ common.register('which', _which, {
 // set on Windows.
 var XP_DEFAULT_PATHEXT = '.com;.exe;.bat;.cmd;.vbs;.vbe;.js;.jse;.wsf;.wsh';
 
+// For earlier versions of NodeJS that doesn't have a list of constants (< v6)
```

Comment:
Could you add a TODO to use the node-provided constant on v6+? We'll eventually drop v4/5.

Reply:
Also, I'm trusting that (1) this value is respected on v4, even though the constant does not exist, and (2) the value is treated the same as `F_OK` on Windows, just as `X_OK` is documented:

> Flag indicating that the file can be executed by the calling process. This has no effect on Windows (will behave like fs.constants.F_OK).

Reply:
True, on Windows `X_OK` and `F_OK` are equivalent. But on Windows platform file extensions are used to determine if file is executable or not.
I will not do this check for Windows platform.

Reply:
> I will not do this check for Windows platform.

Sounds good